src/header/header.scss

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import '../user-icon/index';
+@import '../icon/index';
+
+
+@mixin iui-header {
+  @include iui-reset;
+  @include iui-font-family;
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+  height: $iui-baseline * 5;
+  transition: height $iui-speed-fast ease-out;
+  @include themed {
+    background-color: t(iui-color-background-1);
+    border-bottom: $iui-xxs solid t(iui-color-background-5);
+    color: t(iui-text-color);
+  }
+
+  > .iui-left,
+  > .iui-center,
+  > .iui-right {
+    display: flex;
+    align-items: center;
+    height: 100%;
+  }
+
+  > .iui-left {
+    flex: 2 1 33%;
+    
+    > .iui-application,
+    > .iui-header-button {
+      overflow: hidden;
+      transition: height $iui-speed-fast ease-out;
+
+      > .iui-label {
+        overflow: hidden;
+        text-overflow: ellipsis;
+        transition: font-size $iui-speed-fast ease-out;
+
+        > * {
+          width: 100%;
+          overflow: hidden;
+          text-overflow: ellipsis;
+          text-align: left;
+        }
+      }
+
+      > .iui-brand-logo,
+      > .iui-thumbnail-area {
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        height: 100%;
+
+        > .iui-icon,
+        > .iui-thumbnail {
+          width: $iui-icons-large;
+          height: $iui-icons-large;
+          transition: width $iui-speed-fast ease-out, height $iui-speed-fast ease-out;
+        }
+      }
```

Comment:
I dont know to which extent we want to reuse code or not, but in the HTML example, we use this as the class for a Project button: `iui-button iui-borderless iui-header-button iui-detailed` which means, we alread apply the rules for a `borderless` `button`, in which we expect the child structure to be something like:
```
<iui-button>
  <iui-icon?>
  <iui-label?>
  <iui-icon?>
</iui-button>
```

Now for these buttons, we are adding a very similar structure, but replacing the first `iui-icon` with a container for this `iui-icon`.

I'm not familiar with standalone CSS development, so I dont know to which extent we should consider it, but would it be better to consider `iui-application` a simple override of a button (with a left icon) and `iui-header-button` an "advanced" override of `iui-button iui-dropdown`, In which case `> .iui-brand-logo` and `> .iui-thumbnail-area` could be changed to `iui-application > iui-icon:first-child()` and `iui-header-button > iui-icon:first-child()`, given that it overrides whatever `iui-button > iui-icon` does that we dont want here (if anything)

I think it might be simpler to maintain for the users (for example, having a `<DropdownButton className="iui-header-button" />` might be simpler than to rebuild the same thing, but with a slightly different structure.